### ISSUE TITLE
A 3D Instanseg

### ISSUE THREAD

**Issue by koopa31** [instanseg/instanseg#47]
Hi,

I was wondering if you were planning to implement a 3D version of Instanseg. By this, I mean true 3D segmentation with 3D convolution operations.
At the moment I'm using Cellstitch with Cellpose 2D, which in my experience works better than the 2.5D version of Cellpose, but I'd like to move on to real 3D.
I know that one of the difficult barriers to overcome is the lack of 3D annotated datasets.
Reading the paper it seems that the different losses could be adapted to a 3D groundtruth if I am not mistaken.

PS: During I2K presentation I asked if that could work properly with sparse data. If you are interested in testing the robustness of the method with sparse annotations, you can check Sketchpose paper out. It's a preprint we wrote last year, in which we implemented a simple solution to pseudo-naturally remove annotations from fully annotated datasets and evaluate the performance decrease as a function of the percentage of annotated pixels :). It was nice to see that the performance was not dramatically affected when removing half the annotations.

Thank you for this nice work!

**Comment by ThibautGoldsborough** [comment]
As for the 3d question, initially I thought it would be too messy (see #13 ). But I gave it a go and it looks like there might be a nice workaround that I'm working on. In truth, there are too many different projects that we are working on at the same time and may not be able to do everything in short time. But we are open to help/contributions from the community!

![image](https://github.com/user-attachments/assets/170394e3-2ce9-41e8-b3e9-803ed47b346a)

**Comment by koopa31** [comment]
Indeed, that'd be complex to partially annotate a nuclei without completely changing your implementation. Nevertheless, I think you can easily make data sparse in the sense of getting fully annotated nuclei, but not all of them. That would mean removing some nuclei and parts of the background and label them as -1 I guess.
It's a different project but I find it interesting as you are already trying to use lightweight NN to reduce energy consumption and computation time (I suppose). Hence, showing that your lightweight architecture can still get good performance even when half the nuclei and background are annotated is a cool result.

**Comment by koopa31** [comment]
That's nice, it seems you do not have as many artifacts as for Cellpose. The recombination of 2D masks is a mess and creates a lot of tiny and unrelevant ROIs. You still have some but the result is far more consistent, hence I guess it's a good idea to keep exploring the idea :)

**Comment by koopa31** [comment]
Apart from the difficulty to get some 3D groundtruth, I don't understand very well what would be the real difference in computing 3D or 2D embeddings. Plus, 3D data bring more contextual information tha can be very useful to build relevant feature embeddings. Obviously, I guess there are many subtilities that I don't have in mind as an InstanSeg non-expert ^^

**Comment by ThibautGoldsborough** [comment]
The difficulty is more that the repository was not designed for 3d images from the start. So if you give a 50,256,256 image, InstanSeg will assume that this is a 50 channel image. It sounds silly but rewriting all the "utils" functions (tiling, augmentations etc...) will take me WAY more time than making the loss function work in 3d (which I already more or less did already).

**Comment by koopa31** [comment]
It is true that this represents a lot of work to adapt. 
Claude or ChatGPT can be very useful to earn a lot of time on this kind of silly task like adding cases on dimensions.

**Comment by ThibautGoldsborough** [comment]
Closing this as we're not going to solve this in the near future.
